integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/BoolPredicateSpecificsIT.java

```diff
@@ -99,6 +99,28 @@ public void must() {
 				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_1 );
 	}
 
+	@Test
+	public void and() {
+		assertThatQuery( index.query()
+				.where( f -> f.and( f.match().field( "field1" ).matching( FIELD1_VALUE1 ) )
+				) )
+				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_1 );
+
+		assertThatQuery( index.query()
+				.where( f -> f.and(
+						f.match().field( "field1" ).matching( FIELD1_VALUE1 ),
+						f.match().field( "field2" ).matching( FIELD2_VALUE2 )
+				) ) )
+				.hasNoHits();
+
+		assertThatQuery( index.query()
+				.where( f -> f.and(
+						f.match().field( "field1" ).matching( FIELD1_VALUE1 ),
+						f.match().field( "field2" ).matching( FIELD2_VALUE1 )
+				) ) )
+				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_1 );
+	}
+
```

Comment:
I'd personally prefer that tests for the `AND` predicate be in `AndPredicateSpecificsIT` and tests for the OR predicate be in `OrPredicateSpecificsIT`; it seems more consistent with what we did with other predicates.

Yes it will lead to *some* duplication, but not that much, all things considered. Well worth the cost of a straightforward code structure.

If you disagree, please at least rename the test to make it clear that it's about `bool`, `and` *and* `or`.